myriad-scheduler/src/main/java/org/apache/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -81,83 +83,115 @@ public void onEvent(ResourceOffersEvent event, long sequence, boolean endOfBatch
       }
       return;
     }
-    LOGGER.debug("Received offers {}", offers.size());
+
+    List<Offer> reserved = new ArrayList<>();
+    List<Offer> normals = new ArrayList<>();
+
+    for (Offer offer : offers) {
```

Comment:
You could get fancy here and use sort where comparator does what your isReserved() method is doing.